aips/create_dip.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python
+"""
+Create DIP from AIP
+
+Downloads an AIP from the Storage Service and creates a DIP
+"""
+
+import argparse
+import logging
+import logging.config  # Has to be imported separately
+import os
+import sys
+import requests
+import re
+import tarfile
+import subprocess
+import shutil
+
+THIS_DIR = os.path.abspath(os.path.dirname(__file__))
+LOGGER = logging.getLogger('create_dip')
+
+
+def setup_logger(log_file, log_level='INFO'):
+    """Configures the logger to output to console and log file"""
+    if not log_file:
+        log_file = os.path.join(THIS_DIR, 'create_dip.log')
+
+    CONFIG = {
+        'version': 1,
+        'disable_existing_loggers': False,
+        'formatters': {
+            'default': {
+                'format': '%(levelname)-8s  %(asctime)s  %(filename)s:%(lineno)-4s %(message)s',
+                'datefmt': '%Y-%m-%d %H:%M:%S',
+            },
+        },
+        'handlers': {
+            'console': {
+                'class': 'logging.StreamHandler',
+                'formatter': 'default',
+            },
+            'file': {
+                'class': 'logging.handlers.RotatingFileHandler',
+                'formatter': 'default',
+                'filename': log_file,
+                'backupCount': 2,
+                'maxBytes': 10 * 1024,
+            },
+        },
+        'loggers': {
+            'create_dip': {
+                'level': log_level,
+                'handlers': ['console', 'file'],
+            },
+        },
+    }
+
+    logging.config.dictConfig(CONFIG)
+
+
+def main(ss_url, ss_user, ss_api_key, aip_uuid, tmp_dir):
+    LOGGER.info('Starting DIP creation from AIP: %s', aip_uuid)
+
+    if not os.path.isdir(tmp_dir):
+        LOGGER.error('%s is not a valid directory', tmp_dir)
+        return
+
+    LOGGER.info('Downloading AIP from Storage Service')
+    aip_file = download_aip(ss_url, ss_user, ss_api_key, aip_uuid, tmp_dir)
+
+    if not aip_file:
+        LOGGER.error('Unable to download AIP')
+        return
+
+    LOGGER.info('Extracting AIP')
+    aip_name = extract_aip(aip_file, aip_uuid, tmp_dir)
+
+    if not aip_name:
+        LOGGER.error('Unable to extract AIP')
+        return
+
+    aip_dir = os.path.join(tmp_dir, aip_name)
+
+    LOGGER.info('Creating DIP')
+    dip_dir = create_dip(aip_dir, aip_uuid)
+
+    if not dip_dir:
+        LOGGER.error('Unable to create DIP')
+        return
+
+    LOGGER.info('DIP created in: %s', dip_dir)
+
+
+def download_aip(ss_url, ss_user, ss_api_key, aip_uuid, tmp_dir):
```

Comment:
I'm wondering if this can be done with `amclient`. If it's not possible, maybe you should add it there. @jrwdunham wrote that module and it seems to be very useful. Try https://github.com/artefactual/automation-tools/blob/master/transfers/amclient.py#L555-L581?